### PR TITLE
crafting screen add display co-processing number of remaining operations

### PR DESCRIPTION
**Pull request by asdflj**
crafting screen add display co-processing number of remaining operations

image

### CODE REVIEW

**Comment by Dream-Master** [comment]
@asdflj is this a fix for https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/261 ?

**Comment by asdflj** [comment]
> @asdflj is this a fix for [GTNewHorizons/GT-New-Horizons-Modpack#12724](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12724) ?

nope, only raven know how to fixed

**Comment by asdflj** [comment]
https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12724 this is V2 calculate issue

**Comment by Dream-Master** [comment]
ah ok so a fix we need in 2.3.0

**Comment by Dream-Master** [comment]
@asdflj so this fix a parto of https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12724 ?

**Comment by asdflj** [comment]
nope

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 1]
```diff
@@ -202,6 +271,19 @@ public void drawScreen(final int mouseX, final int mouseY, final float btn) {
         }
 
         super.drawScreen(mouseX, mouseY, btn);
+        this.handleTooltip(mouseX, mouseY, remainingOperations);
+    }
+
+    private void UpdateRemainingOperations() {
```
Comment (repo-alt) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java:277]: why uppercase

File: src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java [hunk 2]
```diff
@@ -1206,6 +1206,14 @@ private TileEntity getTile(ICraftingMedium craftingProvider) {
 
     }
 
+    public int getRemainingOperations() {
+        if (this.isComplete) {
+            return this.accelerator + 1;
```
Comment (repo-alt) [inline src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:1211]: What does it mean? (I mean - why remaining operations of the finished process is not 0?)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java (hunk 1 above)
Comment (asdflj) [reply src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: fixed

File: src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java (hunk 2 above)
Comment (asdflj) [reply src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:1211]: in the ``CraftingCPUCluster::updateCraftingLogic()`` 
``        if (this.waiting || this.tasks.isEmpty()) // nothing to do here...
        {
            return;
        }

        this.remainingOperations = this.accelerator + 1 - (this.usedOps[0] + this.usedOps[1] + this.usedOps[2]);
        final int started = this.remainingOperations;

        if (this.remainingOperations > 0) {
            do {
                this.somethingChanged = false;
                this.executeCrafting(eg, cc);
            } while (this.somethingChanged && this.remainingOperations > 0);
        }
        this.usedOps[2] = this.usedOps[1];
        this.usedOps[1] = this.usedOps[0];
        this.usedOps[0] = started - this.remainingOperations;

        if (this.remainingOperations > 0 && !this.somethingChanged) {
            this.waiting = true;
        }
``
That's why i plus 1

**Comment by chochem** [comment]
we have a feature freeze. can you make this a draft?

**Comment by asdflj** [comment]
> we have a feature freeze. can you make this a draft?

fine,but why is this pr

**Review by repo-alt — APPROVED** [review]

**Comment by repo-alt** [comment]
Why make a draft indeed, I guess Dream can decide by description what needs to be merged and what can be delayed till the next release

**Pull request merged**